File: app/models/payment/CreditCard.java
package models.payment;

import models.BaseModel;
import models.user.User;
import org.joda.time.LocalDate;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import java.util.Date;

@Entity
public class CreditCard extends BaseModel {

    private String ownerName;

    private String number;

    private String code;

    private String creditCardType;

    private LocalDate expirationDate;

    @ManyToOne
    private User user;

    public String getNumber() {
        return number;
    }

    public void setNumber(String number) {
        this.number = number;
    }

    public String getCreditCardType() {
        return creditCardType;
    }

    public void setCreditCardType(String creditCardType) {
        this.creditCardType = creditCardType;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getOwnerName() {
        return ownerName;
    }

    public void setOwnerName(String ownerName) {
        this.ownerName = ownerName;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public LocalDate getExpirationDate() {
        return expirationDate;
    }

    public void setExpirationDate(Date expirationDate) {
        this.expirationDate = LocalDate.fromDateFields(expirationDate);
    }
}

File: ui/src/app/shared/models/payment/credit-card.ts
import {User} from "../user-model";

export class CreditCard {
  public id: string;
  public user: User;
  public ownerName: string = '';
  public number: string = '';
  public code: string = '';
  public expirationDate: Date;
  public creditCardType: string = '';


  constructor(number: string, creditcardtype: string, owner: string, code: string, expirationDate: Date) {
    this.ownerName = owner;
    this.number = number;
    this.code = code;
    this.expirationDate = expirationDate;
    this.creditCardType = creditcardtype;
  }
}
